common/db/tag_store/dynamo.go

```diff
@@ -39,24 +46,33 @@ func (d *DynamoTagStore) Clear() error {
 }
 
 func (d *DynamoTagStore) Insert(tag *models.Tag) error {
+	// ensure we perform a consistent read after a write
+	defer d.setConsistentRead(true)
+
 	tag.TagID = time.Now().UnixNano()
 	return d.table.Put(tag).Run()
 }
 
 func (d *DynamoTagStore) Delete(tagID int64) error {
+	// ensure we perform a consistent read after a delete
+	defer d.setConsistentRead(true)
 	return d.table.Delete("TagID", tagID).Run()
 }
 
 func (d *DynamoTagStore) SelectAll() (models.Tags, error) {
+	defer d.setConsistentRead(false)
+
 	tags := models.Tags{}
-	if err := d.table.Scan().Consistent(true).All(&tags); err != nil {
+	if err := d.table.Scan().Consistent(d.consistentRead).All(&tags); err != nil {
```

Comment:
yeah even with eventually consistent read here, that only halves throughput usage.
our tags table size is getting up to around 10k items with just 12 hours of light usage.

isn't there some way you can query for tags by key type or entity type + environment (i don't see environment id in there right now, but it certainly could be a useful partition or range key.